src/core/issues-notes.js

```diff
@@ -142,10 +135,10 @@ function handleIssues(ins, ghIssues, conf) {
       /** @type {GitHubIssue} */
       let ghIssue;
       if (isIssue) {
-        if (!hasDataNum) {
-          text += ` ${issueNum}`;
-        } else if (dataNum) {
-          text += ` ${dataNum}`;
+        if (report.number !== undefined) {
+          text += ` ${report.number}`;
+        }
+        if ("number" in inno.dataset) {
```

Comment:
Maybe... saves walking the prototype chain (nit pick optimization that's probably not needed).
  
```suggestion
        if (inno.dataset.hasOwnProperty("number")) {
```